src/arrayMethodJoin.js

```diff
@@ -4,8 +4,36 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(initialSeparator) {
+    let joinedArrString = '';
+    let separator = initialSeparator;
+    const newArr = [...this];
+
+    if (separator === undefined) {
+      separator = ',';
+    }
+
+    if (separator === null) {
+      separator = 'null';
+    }
+
+    for (let i = 0; i < newArr.length; i++) {
+      if (newArr[i] === null) {
+        newArr[i] = '';
+      }
+
+      if (newArr[i] === undefined) {
+        newArr[i] = '';
+      }
+
+      joinedArrString += newArr[i];
```

Comment:
You can combine all this logic into one if condition.
Just use `!==` to determine if there is a value and if it is - add it to the string